index.bs

```diff
@@ -2643,6 +2686,33 @@ non-negative integer |defaultBudget|, and an [=aggregatable debug reporting conf
     with |value|, |budget|, |supportedTypes|, and |defaultConfig|.
 1. Return the [=tuple=] (|budget|, |config|).
 
+To <dfn>parse max event states</dfn> from a [=map=] |map| and
+a possibly null 32-bit positive integer |attributionScopeLimit|:
+1. If |attributionScopeLimit| is null:
+    1. If |map|["<code>[=source-registration JSON key/max_event_states=]</code>"] [=map/exists=], return an error.
```

Comment:
Wdyt about allowing default value here? This seems to be better aligned with attribution scopes?

Reply:
we do return default value at line 2711. This line returns an error if attribution scope limit is not set, but max event states is set explicitly. what's the point to add `max_event_states` in json if attribution scope is disabled?

Reply:
It’s just that a default value would be set, but why would it be an error if the default value is set explicitly in the JSON? Similarly we allow empty scopes when attribution scope limit is null.

Reply:
We allow empty scopes when attribution scope limit is null, but we do not allow non-empty scopes when attribution scope is null :)

Reply:
I would say it would be much easier if these fields are wrapped in a dict which requires attribution_scope_limit. 

Currently if we say in the explainer that max_event_states defaults to 3, I feel it a little weird to reject the source when it is explicitly set. 

Cc @apasel422

Reply:
Fair point. Done.